syntax = "proto3";

service TaskService {
  rpc RunTask(TaskRequest) returns (TaskResponse);
  rpc GetTask(Empty) returns (Task);
  rpc TaskResult(TaskResultRequest) returns (Empty);
  rpc GetTaskResult(TaskResultRequest) returns (TaskResultRequest); // Новый метод уведомления клиента
}

message TaskRequest {
  string task_id = 1;
  string task_name = 2;
  repeated string args = 3;
}

message TaskResponse {
  string status = 1;
}

message Task {
  string task_id = 1;
  string task_name = 2;
  repeated string args = 3;
}

message TaskResultRequest {
  string task_id = 1;
  string result = 2;
}

message TaskCompleteNotification {
  string task_id = 1;
  string result = 2;
}

message Empty {}
